slick/src/main/scala/slick/util/InternalArrayQueue.scala

```diff
@@ -0,0 +1,175 @@
+package slick.util
+
+import java.util
+
+class InternalArrayQueue[E >: Null <: AnyRef](capacity: Int) {
```

Comment:
Yes, it is basically the old ManagedArrayBlockingQueue. Maybe we should keep the original name for this class and rename the new ManagedArrayBlockingQueue to PrioritizedBlockingQueue or something?